web/src/components/studies/CreateStudyForm.tsx

```diff
@@ -370,29 +393,37 @@ export default function CreateStudyForm(CreateStudyProps: CreateStudyProps) {
                 Eg. {UclDpoId}/2022/01/123
               </HelperText>
               <div className={styles["data-protection-wrapper"]}>
-                {controllerValue === "UCL" ? (
-                  <Input
-                    type="text"
-                    id="dataProtectionPrefix"
-                    {...register("dataProtectionPrefix")}
-                    readOnly={true}
-                    value={UclDpoId}
-                    inputClassName={styles.readonly}
-                  />
-                ) : (
-                  <Input
-                    type="text"
-                    id="dataProtectionPrefix"
-                    {...register("dataProtectionPrefix")}
-                    placeholder="Registry ID eg ZX1234"
-                  />
-                )}
-
-                <Input type="month" id="dataProtectionDate" {...register("dataProtectionDate")} />
-                <Input
-                  type="number"
-                  id="dataProtectionId"
-                  {...register("dataProtectionId", {
+                <Controller
+                  name="dataProtectionPrefix"
+                  control={control}
+                  render={({ field }) => (
+                    <Input
+                      {...field}
+                      type="text"
+                      id="dataProtectionPrefix"
+                      readOnly={controllerValue === "UCL"}
+                      placeholder={controllerValue === "UCL" ? "" : "Registry ID eg ZX1234"}
+                      inputClassName={controllerValue === "UCL" ? styles.readonly : ""}
+                    />
+                  )}
+                />
+
+                <Controller
+                  name="dataProtectionDate"
+                  control={control}
+                  render={({ field }) => (
+                    <input
```

Comment:
is there a reason this isn't also a UI kit component?

Reply:
For some reason the data was not getting saved when using the uikit components. Happy for you to dig into it to see if you can get it working though.